test/run-drun/upgrade-to-any/version1.mo

```diff
@@ -0,0 +1,24 @@
+import Prim "mo:prim";
+
+actor {
+   Prim.debugPrint("Version 1");
+
+   let allocationSize = 40_000_000;
+
+   func largeAllocation(name: Text): [var Nat] {
+      Prim.debugPrint("Initialize " # name);
+      Prim.Array_init<Nat>(allocationSize / 4, 0);
+   };
+
+   stable var firstVariable : Any = largeAllocation("first variable");
```

Comment:
I know you have special logic to reclaim the space of top-level `Any` fields, but I wonder if we even need that since, I think, your dynamic memory compatibility check allows stable fields to be dropped entirely, which should free the memory. On the other hand, the dfx static check currently doesn't allow them to be dropped, so perhaps the explicit erasure is a good thing (or we should consider relaxing the Motoko stable-compatible static check, used by dfx accordingly).

Reply:
Indeed, this is a very specific optimization, probably not worth it if we would adjust the `dfx` check for canisters with enhanced orthogonal persistence.